### PR TITLE
update all the benchmarks to call pbench-metadata-log directly.

### PR DESCRIPTION
**Pull request by tenstormavi**
FIX of #546

### CODE REVIEW

File: agent/util-scripts/pbench-collect-sysinfo [hunk 1]
```diff
@@ -157,7 +157,7 @@ function gather_remote_sysinfo_data {
 	local remote_label=$2
 	debug_log "[$script_name]running sysinfo-dump on $remote_host"
 	cmd="ssh $ssh_opts -n $remote_host pbench-remote-sysinfo-dump $sysinfo_path "$sysinfo" $remote_label"
-	debug_log "[$script_name] $cmd > $sysinfo_path/$remote_label$remote_host.tar.xz 2> $sysinfo_path/$remote_label$remote_host.err"
+	debug_log "[$script_name] $cmd > $sysinfo_path/$remote_label$remote_host.tar 2> $sysinfo_path/$remote_label$remote_host.err"
```
Comment (portante) [inline agent/util-scripts/pbench-collect-sysinfo]: Looks like you have your work all on one branch.  Let's see if we can tease these out into separate branches and resubmit.

**Comment by tenstormavi** [comment]
@portante @ndokos changes made as requested.

File: agent/bench-scripts/pbench-cyclictest [hunk 2]
```diff
@@ -123,6 +123,7 @@ printf "\n" >>$benchmark_summary_html_file
 export benchmark config
 
 count=1
+pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir beg
```
Comment (portante) [inline agent/bench-scripts/pbench-cyclictest]: We don't need this one, as the `beg` parameter makes it a noop.  This is true for all invocations of `pbench-metadata-log`.

We should remove the `beg` and `end` parameters from `pbench-metadata-log`, modifying that script to stop checking them.

**Comment by ndokos** [comment]
The commit summary includes the unrelated comments from the compressed tarball fix (#544). They should be removed.

**Comment by tenstormavi** [comment]
@portante @ndokos changes made as requested.

Change the script to stop looking for beg and end parameters. But `pbench-collect-sysinfo` also uses those parameters for some functions, how can I deal with them?

**Comment by ndokos** [comment]
So `pbench-collect-sysinfo` should *not* be modified to get rid of the (beg, end) params. I need to take a look at pbench-metadata-log to see if we can get rid of them there - not sure yet.

File: agent/bench-scripts/pbench-cyclictest [hunk 3]
```diff
@@ -160,6 +160,7 @@ done
 if [ $stress ]; then
 	kill $stress_pid
 fi
+pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir
```
Comment (portante) [inline agent/bench-scripts/pbench-cyclictest]: Add "`end`" argument, and invoke it with the "`beg`" argument between lines 126 and 127.

File: agent/bench-scripts/pbench-dbench [hunk 4]
```diff
@@ -511,6 +511,7 @@ for thread_count in `echo $thread_counts | sed -e s/,/" "/g`; do
 done
 
 printf -- "</pre>\n" >>$benchmark_summary_html_file
+pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir
```
Comment (portante) [inline agent/bench-scripts/pbench-dbench]: Add "`end`" argument, and invoke it with the "`beg`" argument between lines 303 and 304.

File: agent/bench-scripts/pbench-fio [hunk 5]
```diff
@@ -886,6 +886,7 @@ mkdir -p $benchmark_run_dir
 export benchmark config
 pbench-collect-sysinfo --group=$tool_group --dir=$benchmark_run_dir beg
 fio_run_benchmark
+pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir
```
Comment (portante) [inline agent/bench-scripts/pbench-fio]: Add "`end`" argument, and invoke it with the "`beg`" argument between lines 887 and 888.

File: agent/bench-scripts/pbench-iozone [hunk 6]
```diff
@@ -185,6 +185,7 @@ done
 
 pbench-stop-tools --group=$tool_group --iteration=$iteration --dir=$benchmark_results_dir
 pbench-postprocess-tools --group=$tool_group --iteration=$iteration --dir=$benchmark_results_dir
+pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir
```
Comment (portante) [inline agent/bench-scripts/pbench-iozone]: Add "`end`" argument, and invoke it with the "`beg`" argument between lines 166 and 167.

File: agent/bench-scripts/pbench-linpack [hunk 7]
```diff
@@ -128,6 +128,7 @@ pushd /usr/local/linpack_11.1.3/benchmarks/linpack
 OMP_NUM_THREADS=$threads ./xlinpack_xeon64 < ./linpack.dat | tee $result_file
 pbench-stop-tools --group=$tool_group --iteration=$iteration --dir=$benchmark_results_dir
 pbench-postprocess-tools --group=$tool_group --iteration=$iteration --dir=$benchmark_results_dir
+pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir
```
Comment (portante) [inline agent/bench-scripts/pbench-linpack]: Add "`end`" argument, and invoke it with the "`beg`" argument between lines 123 and 124.

File: agent/bench-scripts/pbench-migrate [hunk 8]
```diff
@@ -371,6 +371,7 @@ for postprocess_dir in $postprocess_dirs; do
 	pbench-postprocess-tools --group=$tool_group --dir=$postprocess_dir &
 done
 wait
+pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir
```
Comment (portante) [inline agent/bench-scripts/pbench-migrate]: Add "`end`" argument, and invoke it with the "`beg`" argument between lines 332 and 333.

File: agent/bench-scripts/pbench-moongen [hunk 9]
```diff
@@ -480,5 +480,6 @@ for rate in `echo $rates | sed -e s/,/" "/g`; do
 done
 echo "$script_path/postprocess/generate-benchmark-summary $benchmark_name $orig_cmd $benchmark_run_dir" >"$benchmark_run_dir/generate-benchmark-summary.cmd"
 $script_path/postprocess/generate-benchmark-summary "$benchmark_name" "$orig_cmd" "$benchmark_run_dir"
+pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir
```
Comment (portante) [inline agent/bench-scripts/pbench-moongen]: Add "`end`" argument, and invoke it with the "`beg`" argument between lines 339 and 340.

File: agent/bench-scripts/pbench-netperf [hunk 10]
```diff
@@ -588,6 +588,7 @@ for protocol in `echo $protocols | sed -e s/,/" "/g`; do
 	done
 done
 printf -- "</pre>\n" >>$benchmark_summary_html_file
+pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir
```
Comment (portante) [inline agent/bench-scripts/pbench-netperf]: Add "`end`" argument, and invoke it with the "`beg`" argument between lines 408 and 409.

File: agent/bench-scripts/pbench-specjbb2005 [hunk 11]
```diff
@@ -222,6 +222,7 @@ for i in `seq $start_warehouses $inc_warehouses $stop_warehouses`; do
 	pbench-postprocess-tools --group=$tool_group --iteration=$new_iteration_name --dir=$benchmark_results_dir/$new_iteration_name/reference-result
 	let iteration=$iteration+1
 done
+pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir
```
Comment (portante) [inline agent/bench-scripts/pbench-specjbb2005]: Add "`end`" argument, and invoke it with the "`beg`" argument between lines 215 and 216.

File: agent/bench-scripts/pbench-uperf [hunk 12]
```diff
@@ -688,6 +688,7 @@ done
 echo "$script_path/postprocess/generate-benchmark-summary \"$benchmark\" \"$orig_cmd\" \"$benchmark_run_dir\"" >"$benchmark_run_dir/generate-benchmark-summary.cmd"
 chmod +x "$benchmark_run_dir/generate-benchmark-summary.cmd"
 $script_path/postprocess/generate-benchmark-summary "$benchmark" "$orig_cmd" "$benchmark_run_dir"
+pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir
```
Comment (portante) [inline agent/bench-scripts/pbench-uperf]: Add "`end`" argument, and invoke it with the "`beg`" argument between lines 475 and 476.

File: agent/bench-scripts/pbench-user-benchmark [hunk 13]
```diff
@@ -109,10 +109,10 @@ export benchmark_results_dir=$benchmark_results_dir
 # but since we don't call pbench-collect-sysinfo here, we need to call
 # metadata-log explicitly
 export benchmark config
-pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir beg
+pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir
```
Comment (portante) [inline agent/bench-scripts/pbench-user-benchmark]: Restore the previous line.

File: agent/bench-scripts/pbench-user-benchmark [hunk 14]
```diff
@@ -109,10 +109,10 @@ export benchmark_results_dir=$benchmark_results_dir
...
 
 # on abnormal exit, make sure that the metadata log exists and is complete.
-trap "pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir int" INT QUIT
+trap "pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir" INT QUIT
```
Comment (portante) [inline agent/bench-scripts/pbench-user-benchmark]: Restore the previous line, and then add one with "`end`" between lines 123 and 124.

File: agent/util-scripts/pbench-metadata-log [hunk 15]
```diff
@@ -57,9 +57,6 @@ while true; do
 	esac
 done
 
-function usage {
```
Comment (portante) [inline agent/util-scripts/pbench-metadata-log]: Restore this function.

File: agent/util-scripts/pbench-metadata-log [hunk 16]
```diff
@@ -57,9 +57,6 @@ while true; do
...
-    echo "Usage: $prog beg|end <bench_run_dir>" >&2
```
Comment (portante) [inline agent/util-scripts/pbench-metadata-log]: Can you add "`int`" to the list of arguments?

File: agent/util-scripts/pbench-metadata-log [hunk 17]
```diff
@@ -154,7 +151,6 @@ EOF
 function metadata_log_end {
     local dir=$1
     local group=$2
-    local int=$3
```
Comment (portante) [inline agent/util-scripts/pbench-metadata-log]: Restore.

File: agent/util-scripts/pbench-metadata-log [hunk 18]
```diff
@@ -169,30 +165,14 @@ function metadata_log_end {
     if ! grep -q end_run $log ;then
         echo "$(timestamp)" | pbench-add-metalog-option $dir/metadata.log run end_run
     fi
-    if [ "$int" == "int" ] ;then
```
Comment (portante) [inline agent/util-scripts/pbench-metadata-log]: Restore.

File: agent/util-scripts/pbench-metadata-log [hunk 19]
```diff
@@ -169,30 +165,14 @@ function metadata_log_end {
...
-        echo "true"  | pbench-add-metalog-option $dir/metadata.log run run_interrupted
-    fi
 }
 
 mkdir -p $dir
-case $1 in
```
Comment (portante) [inline agent/util-scripts/pbench-metadata-log]: Restore.

File: agent/util-scripts/pbench-metadata-log [hunk 20]
```diff
@@ -169,30 +165,14 @@ function metadata_log_end {
...
-        fi
-        ;;
-    int)
-        metadata_log_end $dir $group int
-        ;;
-    *)
-        usage
-        exit 2
-esac
+metadata_log_end $dir $group
```
Comment (portante) [inline agent/util-scripts/pbench-metadata-log]: Remove.

**Comment by tenstormavi** [comment]
@psuriset @ndokos Made the changes as requested.

**Comment by tenstormavi** [comment]
@psuriset @ndokos @chaitanyaenr changes are made as requested.

File: agent/util-scripts/pbench-metadata-log [hunk 21]
```diff
@@ -58,7 +58,7 @@ while true; do
 done
 
 function usage {
-    echo "Usage: $prog beg|end <bench_run_dir>" >&2
+    echo "Usage: $prog beg|end|int <bench_run_dir>" >&2
```
Comment (ndokos) [inline agent/util-scripts/pbench-metadata-log]: Is "int" used anywhere?

Comment (portante) [inline agent/util-scripts/pbench-metadata-log]: It is used in a trap.

**Pull request merged**